tests/expr_and_series/is_between_test.py

```diff
@@ -22,12 +18,21 @@
     ],
 )
 def test_is_between(constructor: Constructor, closed: str, expected: list[bool]) -> None:
+    data = {"a": [1, 4, 2, 5]}
     df = nw.from_native(constructor(data))
     result = df.select(nw.col("a").is_between(1, 5, closed=closed))
     expected_dict = {"a": expected}
     assert_equal_data(result, expected_dict)
 
 
+def test_is_between_expressified(constructor: Constructor) -> None:
+    data = {"a": [1, 4, 2, 5], "b": [0, 5, 2, 4], "c": [9, 9, 9, 9]}
+    df = nw.from_native(constructor(data))
+    result = df.select(nw.col("a").is_between(nw.col("b"), nw.col("c")))
```

Comment:
Should we test with a generic expression instead of _just_ `col`?
Random proposal that would not change `expected_dict`:

```suggestion
    result = df.select(nw.col("a").is_between(nw.col("b") * 0.9, nw.col("c") - 1))
```